sdk/monitor/monitor-opentelemetry/src/traces/handler.ts

```diff
@@ -171,48 +127,47 @@ export class TraceHandler {
       };
       httpinstrumentationOptions.ignoreOutgoingRequestHook = mergedIgnoreOutgoingRequestHook;
-      this._httpInstrumentation = new HttpInstrumentation(this._config.instrumentationOptions.http);
-      this.addInstrumentation(this._httpInstrumentation);
+      this._instrumentations.push(
+        new HttpInstrumentation(this._config.instrumentationOptions.http)
+      );
     }
-    if (!this._azureSdkInstrumentation) {
-      this._azureSdkInstrumentation = createAzureSdkInstrumentation(
-        this._config.instrumentationOptions.azureSdk
-      ) as any;
-      this.addInstrumentation(this._azureSdkInstrumentation);
+    if (this._config.instrumentationOptions.azureSdk?.enabled) {
+      this._instrumentations.push(
+        createAzureSdkInstrumentation(this._config.instrumentationOptions.azureSdk)
+      );
     }
-    if (!this._mongoDbInstrumentation) {
-      this._mongoDbInstrumentation = new MongoDBInstrumentation(
-        this._config.instrumentationOptions.mongoDb
+    if (this._config.instrumentationOptions.mongoDb?.enabled) {
+      this._instrumentations.push(
+        new MongoDBInstrumentation(this._config.instrumentationOptions.mongoDb)
       );
-      this.addInstrumentation(this._mongoDbInstrumentation);
     }
-    if (!this._mySqlInstrumentation) {
-      this._mySqlInstrumentation = new MySQLInstrumentation(
-        this._config.instrumentationOptions.mySql
+    if (this._config.instrumentationOptions.mySql?.enabled) {
+      this._instrumentations.push(
+        new MySQLInstrumentation(this._config.instrumentationOptions.mySql)
       );
-      this.addInstrumentation(this._mySqlInstrumentation);
     }
-    if (!this._postgressInstrumentation) {
-      this._postgressInstrumentation = new PgInstrumentation(
-        this._config.instrumentationOptions.postgreSql
+    if (this._config.instrumentationOptions.postgreSql?.enabled) {
+      this._instrumentations.push(
+        new PgInstrumentation(this._config.instrumentationOptions.postgreSql)
       );
-      this.addInstrumentation(this._postgressInstrumentation);
     }
-    if (!this._redisInstrumentation) {
-      this._redisInstrumentation = new RedisInstrumentation(
-        this._config.instrumentationOptions.redis
+    if (this._config.instrumentationOptions.redis?.enabled) {
+      this._instrumentations.push(
+        new RedisInstrumentation(this._config.instrumentationOptions.redis)
       );
-      this.addInstrumentation(this._redisInstrumentation);
     }
-    if (!this._redis4Instrumentation) {
-      this._redis4Instrumentation = new Redis4Instrumentation(
-        this._config.instrumentationOptions.redis4
+    if (this._config.instrumentationOptions.redis4?.enabled) {
+      this._instrumentations.push(
+        new Redis4Instrumentation(this._config.instrumentationOptions.redis4)
       );
-      this.addInstrumentation(this._redis4Instrumentation);
     }
+    // this.disableInstrumentations = registerInstrumentations({
```

Comment:
Were you intending to add a method to disable instrumentations individually here? Or is this not going to be included?

Reply:
Already included below, removing duplicated code